src/generators/schema-generator.php

```diff
@@ -130,7 +130,80 @@ public function generate( Meta_Tags_Context $context ) {
 	}
 
 	/**
-	 * Allow filtering the graph piece by its schema type.
+	 * Adapts the WebPage graph piece for password-protected posts.
+	 *
+	 * It should only have certain whitelisted properties.
+	 * The type should always be WebPage.
+	 *
+	 * @param array $graph_piece The WebPage graph piece that should be adapted for password-protected posts.
+	 *
+	 * @return array The WebPage graph piece that has been adapted for password-protected posts.
+	 */
+	public function protected_webpage_schema( $graph_piece ) {
+		$properties_to_show = \array_flip(
+			[
+				'@type',
+				'@id',
+				'url',
+				'name',
+				'isPartOf',
+				'inLanguage',
+				'datePublished',
+				'dateModified',
+				'breadcrumb',
+			]
+		);
+
+		$graph_piece          = \array_intersect_key( $graph_piece, $properties_to_show );
```

Comment:
can $graph_piece be null or empty? is it still okay if no matches are found?

Reply:
Goog point.

Reply:
To fix this i've made it so that our add filter runs first.